### PR TITLE
[FLASK] Allow Snaps to use `eth_accounts` as a revokable permission

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Explanation

Allows Snaps to access `eth_accounts` as a revokable permission. This means that snaps now can use `eth_accounts` and `eth_requestAccounts` to prompt the user for account selection. This permission can be revoked again in the snaps settings whenever.

This PR is mainly UI changes to make the account connection flow work with these changes and UI additions for permission revoking.

This PR needs the latest, unpublished version of the snaps monorepo to work completely.

### CODE REVIEW

File: ui/components/app/permission-cell/permission-cell.js
Comment (hmalik88) [inline ui/components/app/permission-cell/permission-cell.js:1]: Shouldn't the additions here be fenced?

Comment (FrederikBolding) [reply ui/components/app/permission-cell/permission-cell.js:1]: The permission cell component is currently only used in snaps screens, so I don't think that's necessary.

File: ui/pages/permissions-connect/permissions-connect.component.js [hunk 1]
```diff
@@ -268,30 +277,57 @@ export default class PermissionConnect extends Component {
...
-          >
-            <Icon
-              name={IconName.ArrowLeft}
-              marginInlineEnd={1}
-              size={IconSize.Xs}
+      <Box
+        style={{
+          ///: BEGIN:ONLY_INCLUDE_IN(snaps)
+          marginBottom:
+            targetSubjectMetadata.subjectType === SubjectType.Snap && '14px',
```
Comment (GuillaumeRx) [inline ui/pages/permissions-connect/permissions-connect.component.js:293]: this should be moved out of `style` in favour of Box `marginBottom` own prop and use either `2` or `4` for the value

Comment (GuillaumeRx) [reply ui/pages/permissions-connect/permissions-connect.component.js:293]: After looking at it it seems like with the legacy style it's too much pain to make this change.

**Review by GuillaumeRx — APPROVED** [review]

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**